src/components/structures/MessagePanel.tsx

```diff
@@ -52,8 +53,10 @@ import EditorStateTransfer from "../../utils/EditorStateTransfer";
 import { Action } from '../../dispatcher/actions';
 import { getEventDisplayInfo } from "../../utils/EventUtils";
 import { IReadReceiptInfo } from "../views/rooms/ReadReceiptMarker";
+import UIStore, { UI_EVENTS } from '../../stores/UIStore';
 
 const CONTINUATION_MAX_INTERVAL = 5 * 60 * 1000; // 5 minutes
+const NARROW_MODE_BREAKPOINT = 400;
```

Comment:
Feels very bizarre that the composer and timeline have different narrow mode breakpoints (500 and 400 respectively)

So the UI would pop multiple times as you shrink/expand it, can we not consolidate this logic to a parent of them both and settle on one width?

Reply:
Yes, I noticed the separate breakpoint for the composer as well... 

I suppose it largely depends on whether @janogarcia and other designers think those components should adapt at the same breakpoint or not. The composer is a bit wider than timeline entries (at least if you are showing read receipts on the right), so I can see an argument that they may have different breakpoints.

Reply:
Agreed, but the jarring-ness of multiple UI "jumps" should also be considered

Reply:
At the moment at least, the composer breakpoint only hides a single button (upload) when narrow, which (to me at least) is hard to notice visually unless you know to look for it.

In general though, I agree it could eventually get out of hand if we had lots of components all changing at distinct widths.

Reply:
@t3chguy @jryans Unsure if a global breakpoint will work for every single component in the timeline (current and future), but we can test it and adjust. 👍 We won't know that until we invest the necessary time for reviewing the responsiveness of the timeline properly. 

That said, the composer breakpoint is probably not used anymore, given that we recently changed its behavior to collapsed by default.

Reply:
> That said, the composer breakpoint is probably not used anymore, given that we recently changed its behavior to collapsed by default.

It still is, the breakpoint hides the upload button into the overflow menu

Reply:
Thanks! Forgot about that detail. 😄 Anyway, let's just reuse the same breakpoint for now. 👍